libutils/file_lib.c

```diff
@@ -453,6 +454,13 @@ int safe_open(const char *pathname, int flags, ...)
                 return -1;
             }
 
+            // We can't safely open(2) a FIFO without blocking infinitely
+            if (S_ISFIFO(stat_before.st_mode))
+            {
+                close(currentfd);
+                return -1;
+            }
```

Comment:
I think this should be handled elsewhere, since it is not categorically wrong to open a fifo, it's just in the particular way we have been using it from our ``chown`` implementation. In fact, is this even needed anymore? The new implementations of ``safe_chmod``, ``safe_chown`` and ``safe_lchown`` won't open files anymore.

Opening fifos in general could be desirable, for example using ``readfile`` from policy, when you know that another process is at the other end.